Improve ESLint configuration

gbirke opened the pull request:
Use the new "flat" configuration style, copied over from the fundraising application.

Use the newly-converted "wikimedia" rules for formatting and typescript.

Apply fixes

- [x] Style code
- [x] Add Vue rules from wikimedia config
- [x] Update to ESLint 9 and clean up ESLint dependencies

Abban commented on thank_you/components/ProgressBar.jsx:
```diff
@@ -1,4 +1,3 @@
-// eslint-disable-next-line no-unused-vars
 import { h } from 'preact';
 // This component uses Component from compat because pure Preact does not support
```
IIRC all the new banners use CSS animation for the progress bar, so we might be able to remove compat

gbirke committed:
Use new ESLint configuration

Use the new "flat" configuration style, copied over from the fundraising
application.

Use the newly-converted "wikimedia" rules for formatting and typescript.

Fix all files where ESLint had a legitimate complaint, applying rules
that were not there before and that are not auto-fixable or just
formatting issues.

gbirke committed:
Add Vue ESLint rules

Copied the rules from the [mediawiki coding style](https://github.com/wikimedia/eslint-config-wikimedia)

Marked & commented out rules that will change the code significantly,
those will need a discussion & decision with the whole team.

Commented out some rules that'll be part of the commit that
automatically fixes violations.

Fixed/changed some components that had violations. The components in
DonationForm/Forms all have an `isCurrent` property that's no longer
used in the markup. I decided to ignore the notification for now,
because the banners component still pass `is-current` to the forms and
the property could be useful in the future.

gbirke committed:
Add Vue formatting rules

The formatting rules are purely stylistic and add more spaces in Vue components.

Auto-Fix components

gbirke committed:
Activate auto-fixable ESLint rules

Activate auto-fixable ESLint rules and fix code style

gbirke committed:
Update to ESLint 9

Remove eslint-config-wikimedia - we have implemented its rules in the
ESLint 9 "flat" configuration ourselves

Remove eslint-plugin-jsdoc and eslint-plugin-yml - we're not using them
and they were probably an attempt to fix a eslint-config-wikimedia
dependency

Add types to MultiStepDonation form component to make the update
TypeScript compiler happy
